pkg/execution/executor/executor.go

```diff
@@ -652,6 +690,29 @@ func (e *executor) Execute(ctx context.Context, id state.Identifier, item queue.
 
 		// Only just starting:  run lifecycles on first attempt.
 		if item.Attempt == 0 {
+			// NOTE:
+			// annotate the step as the first step of the function run.
+			// this way the delay associated with this run is directly correlated to the delay of the
+			// function run itself.
+			span.SetAttributes(attribute.Bool(consts.OtelSysStepFirst, true))
```

Comment:
annotate the step as the first step so we can infer that the latency this step is the same as the function run latency.

Reply:
Does this mean we can only infer latency for the very first step?

I'm wondering how we would infer latency for parallel steps?

Reply:
System/sojourn delay can help us with parallel step latency.